### ISSUE TITLE
Sidecar interferes with drive locking / ejecting

### ISSUE THREAD

**Issue by Vanhecke** [Graylog2/collector-sidecar#146]
### Problem description
I was trying to write an image to USB (Win32DiskImager) and kept getting "Error 5 Access Denied".
### Steps to reproduce the problem

1. Run sidecar (with nx backed in my case)
2. Input USB
3. Run win32diskimager
4. try to flash something

This tool helped me find the culprit: http://safelyremove.com/index.htm

### Environment

* Sidecar Version: Pretty sure its the latest - I installed it 23/2. Don't see a version number...
* Graylog Version: N/A
* Operating System: 10 Enterprise - latest
* Elasticsearch Version: N/A
* MongoDB Version:N/A

**Comment by mariussturm** [comment]
Hi,
the Sidecar itself is not locking or opening many files. Basically just a couple of log files in C:\\Program Files\Graylog\collector-sidecar. Maybe you configured one of the backends like nxlog or filebeat to open a file on a USB drive?

**Comment by mariussturm** [comment]
@Vanhecke any news on this?

**Comment by Vanhecke** [comment]
@mariussturm 
I know that the reason why I couldn't eject a device was because of the graylog-collector-sidecar.exe, not the nxlog service.
It hasn't happened since so I'll close this.
Will open it again if I can reproduce.

**Comment by mariussturm** [comment]
Alright, when you have a reproducible scenario feel free to re-open!

**Comment by g-t-u** [comment]
Hi, I would like that this issue to be re-opened, but I'm new to github and can't find if I can do this.
I gathered more information and steps to reproduce a problem.

### Problem description

One cannot 'Eject safely' removable drive on a Windows host if colelctor-sidecar service is running.

### Steps to reproduce the problem

1. Install collector sidecar and start a service.
2. Insert a removable drive (e.g. a flash thumbdrive).
3. Leave it inserted for a period langer than set in update_interval configuration parameter.
4. When you try to eject thumbdrive you get a message that it cannot be ejected safely.
5. Stop collector-sidecar service, then you can eject a removable drive.

collector_sidecar.yml:
```
server_url: https://some.host:9043/api/
update_interval: 10
tls_skip_verify: false
send_status: true
list_log_files:
node_id: ${sidecar_node_id:graylog_collector_sidecar}
collector_id: file:C:\Program Files\graylog\collector-sidecar\collector-id
cache_path: C:\Program Files\graylog\collector-sidecar\cache
log_path: C:\Program Files\graylog\collector-sidecar\logs
log_rotation_time: 86400
log_max_age: 604800
tags: ["windowswks"]
backends:
    - name: nxlog
      enabled: false
      binary_path: C:\Program Files (x86)\nxlog\nxlog.exe
      configuration_path: C:\Program Files\graylog\collector-sidecar\generated\nxlog.conf
    - name: winlogbeat
      enabled: false
      binary_path: C:\Program Files\graylog\collector-sidecar\winlogbeat.exe
      configuration_path: C:\Program Files\graylog\collector-sidecar\generated\winlogbeat.yml
    - name: filebeat
      enabled: false
      binary_path: C:\Program Files\graylog\collector-sidecar\filebeat.exe
      configuration_path: C:\Program Files\graylog\collector-sidecar\generated\filebeat.yml
```
As you can mention, this bahaviour doesn't depend on which backends are enabled.
I checked activity of a process Graylog-collector-sidecar.exe using utilities Process Explorer and Process Monitor from Sysinternals Suite:

1. At any given moment there are 1 to 12 file handles open for a root directory ov every availablle drive.
2. Every update_interval seconds additional handle is opened. (Porcess Monitor log show that driectory handle is opened twice and closed once).
3. Every 12th cycles all left-over handles are closed an new one is opened.

I'm not an expert in Go language and its runtime, but I suppose tha the cause of this behaviour is in getWindowsDrives function in common/sigar.go module. There os.Open() call has no corresponding .Close() call.

### Environment

* Sidecar Version: 0.1.1
* Graylog Version: 2.2.3
* Operating System: Windows 7
* Elasticsearch Version: doesn't matter
* MongoDB Version: doesn't matter

**Comment by mariussturm** [comment]
Ah, that could actually be. Could you please test the following dev build (needs to be unzipped before!):

[graylog-collector-sidecar.exe.gz](https://github.com/Graylog2/collector-sidecar/files/1098627/graylog-collector-sidecar.exe.gz)

I have add a `Close()` to the drive lookup function to prevent file handle leak. Good catch!

**Comment by g-t-u** [comment]
I've tested this build. File handles are not leaking anymore.

**Comment by mariussturm** [comment]
Just released 0.1.3 with the applied fix. Thanks for reporting all!
